notification/events.go

```diff
@@ -274,7 +275,27 @@ func sendNotifications(ctx context.Context, events models.Events) models.Events
 			e.SetError(err.Error())
 			failedEvents = append(failedEvents, e)
 			notificationContext.WithError(err.Error())
-		} else if err := PrepareAndSendEventNotification(notificationContext, payload, celEnv); err != nil {
+		}
+
+		if e.Delay != nil {
+			// This was a delayed notification.
+			// We need to re-evaluate the health of the resource.
+
+			health, err := celEnv.GetResourceHealth(ctx)
+			if err != nil {
+				e.SetError(err.Error())
+				failedEvents = append(failedEvents, e)
+				notificationContext.WithError(err.Error())
+				continue
+			}
+
+			if health != api.EventToHealth(originalEvent.Name) {
```

Comment:
Need to cater for a notifiucation on warning/unhealthy and the eveny firing on warning and then changing to the other